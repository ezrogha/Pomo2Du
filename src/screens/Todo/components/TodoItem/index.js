import React from 'react';
import { Text, View, Dimensions, StyleSheet, TouchableOpacity } from 'react-native';
import { MaterialCommunityIcons, Ionicons } from '@expo/vector-icons';

const { width } = Dimensions.get('window');

export default ({ data: { title, isChecked, id }, toggleCheckbox, segue}) => (
    <TouchableOpacity activeOpacity={0.9} onPress={segue}>
        <View style={styles.todoContainer}>
            <View style={{ flexDirection: "row", }}>
                <MaterialCommunityIcons onPress={() => toggleCheckbox(id, !isChecked)} name={isChecked ? "checkbox-marked-outline" : "checkbox-blank-outline"} color="green" size={32} />
                <View style={styles.todoTextContainer}>
                    <Text style={styles.todoText}>{title}</Text>
                </View>
            </View>
            <View style={styles.iconContainer}>
                <Ionicons name="ios-arrow-up" color="green" size={25} />
                <View style={{ width: 10 }} />
                <Ionicons name="ios-arrow-down" color="green" size={25} />
            </View>
        </View>
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    todoContainer: {
        justifyContent: 'space-between',
        flexDirection: "row",
        width: width - 20,
        borderBottomWidth: 1,
        borderBottomColor: 'grey',
        alignSelf: 'center',
        marginBottom: 20
    },
    todoText: {
        fontWeight: 'bold',
        fontSize: 20,
        color: 'grey'
    },
    todoTextContainer: {
        justifyContent: 'center',
        marginLeft: 10,
    },
    iconContainer: {
        justifyContent: 'center',
        flexDirection: 'row',
        marginHorizontal: 10,
        paddingTop: 4
    }
})