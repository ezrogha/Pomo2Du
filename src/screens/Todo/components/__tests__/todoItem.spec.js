import React from 'react'

import TodoItem from '../TodoItem';
import renderer from 'react-test-renderer'

describe("Testing TodoItem Component", () => {
    test("snapshot test", () => {
        const props = {
            data: {
                title: "Title",
                isChecked: true,
                id: 1,
                isRunning: true
            },
            toggleCheckbox: () => { },
            segue: () => { },
            deleteTodo: () => { },
            startTask: () => {},
            xTodoValue: 1
        }
        const tree = renderer.create(<TodoItem {...props} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
