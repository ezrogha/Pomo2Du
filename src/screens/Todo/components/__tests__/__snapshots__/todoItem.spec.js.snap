// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TodoItem Component snapshot test 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "borderBottomColor": "grey",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 1,
        "marginBottom": 20,
        "width": 730,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <View
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Title
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 10,
          "paddingTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "width": 20,
          }
        }
      />
      <Text />
    </View>
  </View>
</View>
`;
